run-tests.ps1

```diff
@@ -58,26 +74,36 @@ foreach ($framework in $testFrameworks) {
         continue
     }
 
-    Write-Output "::group::Running test suite for $framework on $platform"
-    # -m:1 parameter prevents test projects from being run in parallel, which could cause conflicts since PessimisticLocks project shares same tests
-    dotnet test src --configuration Release --no-build --framework $framework --logger "GitHubActions;report-warnings=false" -m:1
+    foreach ($project in $testProjects) {
 
-    if ($LASTEXITCODE -ne 0) {
-        $exitCode = 1
-    }
+        if (-not $project.Value.Contains($framework)) {
+            Write-Output "Skipping $(Split-Path $project.Name -leaf) ($framework on $platform)"
+            continue
+        }
 
-    Write-Output "::endgroup::"
+        Write-Output "::group::Running $(Split-Path $project.Name -leaf) ($framework on $platform)"
+
+        # -m:1 parameter prevents test projects from being run in parallel, which could cause conflicts since PessimisticLocks project shares same tests
+        dotnet test $project.Name --configuration Release --no-build --framework $framework --logger "GitHubActions;report-warnings=false" -m:1
```

Comment:
The "PessimisticLocks" comment was a specific thing in Raven, so kind of funny I forgot it here. But if we're doing these one at a time anyway, isn't `-m:1` then pointless?

```suggestion
        dotnet test $project.Name --configuration Release --no-build --framework $framework --logger "GitHubActions;report-warnings=false"
```